<a href="https://colab.research.google.com/github/rsurapol/Python-Programming/blob/master/Chapter3_Variable.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# บทที่ 3 ตัวแปร ชนิดข้อมูล นิพจน์ และตัวดำเนินการ

## การประกาศตัวแปร (Declaration)

In [ ]:
#ex-3.1
ปากกา = "สีแดง"
โทรศัพท์ = "Samsung"
print ("ปากกา สี =", ปากกา)
print ("โทรศัพท์ ยี่ห้อ =", โทรศัพท์)

## ตัวแปรชนิดค่าคงที่ (Constant)

In [ ]:
#ex-3.2
pi = 3.14
area = pi * 5 * 5
circumference = 2 * pi * 10
print ("พื้นที่วงกลม =", area)
print ("เส้นรอบวงกลม =", circumference)

## ชนิดข้อมูลจำนวนเต็ม (Integer)

In [ ]:
#ex-3.3
num1 = 546135478945123456214; num2 = 14567489412457845679
print("ผลบวกของ num1 กับ num2 =", num1 + num2)
print("ผลลบของ num1 กับ num2 =", num1 - num2)

In [ ]:
#ex-3.4 การแปลงเลขฐานสิบเป็น ฐานสอง ฐานแปด ฐานสิบหก
x = 6749
print ("แปลงเลขฐานสิบเป็นฐานสอง =", bin(x))
print ("แปลงเลขฐานสิบเป็นฐานแปด =", oct(x))
print ("แปลงเลขฐานสิบเป็นฐานสิบหก =", hex(x))

In [ ]:
#ex-3.5 การแปลงเลขฐานสอง ฐานแปด ฐานสิบหก เป็นฐานสิบ
a = 0b100111011001
b = 0o6453
c = 0xAC41
print ("แปลงเลขฐานสองเป็นฐานสิบ =", int(a))
print ("แปลงเลขฐานแปดเป็นฐานสิบ =", int(b))
print ("แปลงเลขฐานสิบหกเป็นฐานสิบ =", int(c))

In [ ]:
#ex-3.6 การบวกเลขฐานสองและการแปลงผลลัพธ์จากเลขฐานสิบ เป็นฐานสอง ฐานแปด ฐานสิบหก
a = 0b1001011; b = 0b1101101
c = a + b
print ("ผลบวกของ a กับ b =", c)
print ("แปลงผลลัพธ์จากการผลบวก a กับ b เป็นเลขฐานสอง =", bin(c))
print ("แปลงผลลัพธ์จากการผลบวก a กับ b เป็นเลขฐานแปด =", oct(c))
print ("แปลงผลลัพธ์จากการผลบวก a กับ b เป็นเลขฐานสิบหก =", hex(c))

## ชนิดข้อมูลจำนวนทศนิยม (Float)

In [ ]:
#ex-3.7 การแสดงผลการบวก ลบ กับชนิดข้อมูลจำนวนทศนิยม
a = 341.451E-3; b = 251.147e-2; c = 758.457E5; x = 10.5; y = 17.5
d = a - b
z = x + y
print ("ค่าตัวเลขจำนวนทศนิยมของ b = ", b)
print ("ผลลัพธ์จาการลบ a กับ b =", d)
print ("ผลลัพธ์จากการบวก x กับ y =", z)

In [1]:
#ex-3.8 การแปลงชนิดข้อมูลอักขระหรือสตริง เป็นชนิดข้อมูลจำนวนทศนิยม
a = 56; b = 55
c = float(a)
d = float(b)
x = c + d
y = c * d
print ("ผลลัพธ์จากการแปลงค่า 56 =", c)
print ("ผลลัพธ์จาการแปลงค่า 55 =", d)
print ("ผลลัพธ์จาการบวกค่าของ c + d =", x)
print ("ผลลัพธ์จากการคูณค่าของ c * d =", y)

ผลลัพธ์จากการแปลงค่า 56 = 56.0
ผลลัพธ์จาการแปลงค่า 55 = 55.0
ผลลัพธ์จาการบวกค่าของ c + d = 111.0
ผลลัพธ์จากการคูณค่าของ c * d = 3080.0


## ชนิดข้อมูลจำนวนเชิงซ้อ (Complex)

In [ ]:
#ex-3.9 การใช้ตัวดำเนินการบวก ลบ กับชนิดข้อมูลเชิงซ้อน
a = 20 + 5j; b = 15 + 6j; c = 6 - 7j
x = a + b
y = b * c
print ("ผลลัพธ์การบวกระหว่าง a กับ b =", x)
print ("ผลลัพธ์การคูณระหว่าง b กับ c =", y)

## ชนิดข้อมูลตรรกะ (Boolean)

In [2]:
#ex-3.10 การเขียนคำสั่งโปรแกรมใช้งานชนิดข้อมูลตรรกะ
x = True; y = False; z = True
print("ผลการเปรียบเทียบ x == y หรือไม่ =", x == y)
print("ผลการเปรียบเทียบ y == z หรือไม่ =", y == z)
print("ผลการเปรียบเทียบ z == True หรือไม่ =", z == True)

ผลการเปรียบเทียบ x == y หรือไม่ = False
ผลการเปรียบเทียบ y == z หรือไม่ = False
ผลการเปรียบเทียบ z == True หรือไม่ = True


In [ ]:
#ex-3.11 การใช้งานฟังก์ชัน bool() ตรวจสอบค่าข้อมูลในตัวแปร
b = ""; c = " "; n = None; num = 245
print ("มีข้อมูลอยู่ในตัวแปร b หรือไม่ =", bool(b))
print ("มีข้อมูลอยู่ในตัวแปร c หรือไม่ =", bool(c))
print ("มีข้อมูลอยู่ในตัวแปร n หรือไม่ =", bool(n))
print ("มีข้อมูลอยู่ในตัวแปร num2 หรือไม่ =", bool(num))

## ชนิดข้อมูลสายอักขระหรือสตริง (String)

In [ ]:
#ex-3.12 การประกาศสร้างตัวแปรชนิดข้อมูลสตริง
msg = "Python Programming"
msg1 = 'Language'
print(msg)
print(msg1)
print(msg, msg1)

## การแปลงชนิดข้อมูล (Data Type Conversion)

In [ ]:
#ex-3.13 การแจ้งเตือนข้อผิดพลาดเมื่อนำชนิดข้อมูลสตริงดำเนินการกับชนิดข้อมูลจำนวนเต็ม
num = input("Enter number = ")
print(50 + num)

In [ ]:
#ex-3.14 การแปลงชนิดข้อมูลสตริงเป็นชนิดข้อมูลจำนวนเต็ม
num = int(input("กรุณาป้อนตัวเลขของคุณ : "))
x = "60"
y = int(x) + num
print ("ผลลัพธ์จากการบวก x + num =", y)

In [ ]:
#ex-3.15 การแปลงชนิดข้อมูลสตริงเป็นชนิดข้อมูลจำนวนทศนิยม
flo = float(input("กรุณาป้อนตัวเลขของคุณ : "))
x = "152.124"
y = float(x) + flo
print ("ผลลัพธ์จากการบวก x + flo =", y)

In [ ]:
#ex-3.16 การแปลงชนิดข้อมูลสตริงเป็นชนิดข้อมูลเชิงซ้อน
a = "20+5j"; b = "15+6j";
x = complex(a)
y = complex(b)
z = complex(7,5)
print ("ผลลัพธ์การแปลงชนิดข้อมูลจำนวนเชิงซ้อนจากตัวแปร a =", x)
print ("ผลลัพธ์การแปลงชนิดข้อมูลจำนวนเชิงซ้อนจากตัวแปร b =", y)
print ("ค่าจำนวนจริงของ c =", x.real, "และค่าจินตภาพของ c = ", x.imag)
print ("ผลลัพธ์การแปลงเครื่องหมายตัวดำเนินการของตัวแปร y =", y.conjugate())
print ("ผลลัพธ์การสร้างคู่อันดับจากตัวแปร z =", z)

In [3]:
#ex-3.17 การแปลง สตริง เป็น จำนวนเชิงซ้อน กรณีเกิดการแจ้งเตือนข้อผิดพลาด ที่มีช่องว่าง
c = "25 + 9j";
print(complex(c))

ValueError: ignored

## การกำหนดค่าให้กับตัวแปร

In [ ]:
#ex-3.18 การสร้างตัวแปรครั้งละหลายๆ ตัว และกำหนดเป็นชนิดข้อมูลเดียวกัน
num_1 = num_2 = num_3 = 50
print ("ค่าข้อมูลในตัวแปร num_1 =", num_1)
print ("ค่าข้อมูลในตัวแปร num_2 =", num_2)
print ("ค่าข้อมูลในตัวแปร num_3 =", num_3)

In [ ]:
#ex-3.19 การสร้างตัวแปรครั้งละหลายๆ ตัว และกำหนดชนิดข้อมูลจำนวนเต็ม และสตริง 
num1, num2, msg = 50, 100, "Hello World"
print ("ค่าข้อมูลในตัวแปร num1 =", num1)
print ("ค่าข้อมูลในตัวแปร num2 =", num2)
print ("ค่าข้อมูลในตัวแปร msg = ", msg)

In [ ]:
#ex-3.20 การสร้างตัวแปรครั้งละหลายๆ ตัว แต่แยกเก็บชนิดข้อมูลอยู่ในบรรทัดเดียวกัน
x = 200; y = 300; msg = "Hello Python"
print ("ค่าข้อมูลในตัวแปร x =", x)
print ("ค่าข้อมูลในตัวแปร y =", y)
print ("ค่าข้อมูลในตัวแปร msg = ", msg)

## การตรวจสอบชนิดข้อมูล

In [ ]:
#ex-3.21 การตรวจสอบชนิดข้อมูลด้วยฟังก์ชัน type()
msg = "Python Programming"
num = 254
float_ = 354.213
print(type(msg))
print(type(num))
print(type(float_))

## นิพจน์ (Expressions)

---
นิพจน์ เป็นการดำเนินการทางคณิตศาสตร์เพื่อคำนวณ หรือ ทำการเปรียบเทียบหาค่าต่างๆ ตามต้องการ ในการดำเนินการจะประกอบด้วย ค่าคงที่ หรือ ตัวแปร สิ่งเหล่านี้เรียกว่า 

*   ตัวถูกดำเนินการ (Operand)
*   ตัวดำเนินการ (Operator) คือ สัญลักษณ์ต่างๆ เช่น +,-,*,/,<,>,= เป็นต้น






## ตัวดำเนินการทางคณิตศาสตร์ (Arithmetic Operator)

In [ ]:
#ex-3.22 การนำตัวดำเนินการมาใช้
a = 5; b = 3;
z = a + b
print("ผลลัพธ์ของ a + b คือ ", z)
z = a - b
print("ผลลัพธ์ของ a - b คือ ", z)
z = a * b
print("ผลลัพธ์ของ a * b คือ ", z)
z = a / b
print("ผลลัพธ์ของ a / b คือ ", z)
z = a % b
print("ผลลัพธ์ของ a % b คือ ", z)
z = a ** b
print("ผลลัพธ์ของ a ** b คือ ", z)
z = a // b
print("ผลลัพธ์ของ a // b คือ ", z)

## ตัวดำเนินการเปรียบเทียบ (Comparison Operator)

In [ ]:
#ex-3.23 การใช้ตัวดำเนินการเปรียบเทียบ
a = 5; b = 3; c = 15; d = 3
w = a == c
print("a เท่ากับ c หรือไม่ =", w)
x = d <= a
print("d น้อยกว่าหรือเท่ากับ a หรือไม่ =", x)
y = b >= c
print("b มากกว่าหรือเท่ากับ c หรือไม่ =", y)
z = b == d < a
print("b เท่ากับ d และน้อยกว่า a หรือไม่ =", z)

## ตัวดำเนินการกำหนดค่า (Assignment Operator)

In [ ]:
#ex-3.24-1
a = 5; b = 3
a = b
print("ผลลัพธ์ a = b :", a)

In [ ]:
#ex-3.24-2
a = 5; b = 3
a += b
print("ผลลัพธ์ a += b :", a)

In [ ]:
#ex-3.24-3
a = 5; b = 3
a -= b
print("ผลลัพธ์ a -= b :", a)

In [ ]:
#ex-3.24-4
a = 5; b = 3
a *= b
print("ผลลัพธ์ a *= b :", a)

In [ ]:
#ex-3.24-5
a = 5; b = 3
a /= b
print("ผลลัพธ์ a /= b :", a)

In [4]:
#ex-3.24-6
a = 5; b = 3
a %= b
print("ผลลัพธ์ a %= b :", a)

ผลลัพธ์ a %= b : 2


In [5]:
#ex-3.24-7
a = 5; b = 3
a **= b
print("ผลลัพธ์ a **= b :", a)

ผลลัพธ์ a **= b : 125


In [6]:
#ex-3.24-8
a = 5; b = 3
a //= b
print("ผลลัพธ์ a //= b :", a)

ผลลัพธ์ a //= b : 1


## ตัวดำเนินการตรรกศาสตร์ (Logical Operator)

In [ ]:
#ex-3.25
a = 5; b = 3; c = 15; z = 0

z = a == b and c < b
print("ผลลัพธ์ของ a == b and c < b คือ ", z)
z = a > b and a < c
print("ผลลัพธ์ของ a > b and a < c คือ ", z)
z = b > a or a > c
print("ผลลัพธ์ของ b > a or a > c คือ ", z)
z = b < a or c > a
print("ผลลัพธ์ของ b < a or c > a คือ ", z)
z = not (a < c)
print("ผลลัพธ์ของ not (a < c) คือ ", z)
z = not (b > a or c < a)
print("ผลลัพธ์ของ not (b > a or c < a) คือ ", z)

## ตัวดำเนินการระดับบิต (Bitwise Operator)

In [17]:
#ex-3.26
a = 5; b = 3; c= 5
"""
แปลงเป็นเลขฐาน 2 เพื่อความเข้าใจ
ab=bin(a); bb=bin(b); cb=bin(c)
print("5 = ",ab)
print("3 = ",bb)
print("5 = ",cb)
"""
z = a & b
print("ผลลัพธ์ของ a & b คือ ", z)
z = a ^ c
print("ผลลัพธ์ของ a ^ c คือ ", z)
z = a | b
print("ผลลัพธ์ของ a | b คือ ", z)
z = a << 1
print("ผลลัพธ์ของ a << 1 คือ ", z)

#print(int(0b1010))

ผลลัพธ์ของ a & b คือ  1
ผลลัพธ์ของ a ^ c คือ  0
ผลลัพธ์ของ a | b คือ  7
ผลลัพธ์ของ a << 1 คือ  10


## ตัวดำเนินการตรวจสอบสมาชิก (Membership Operator)

In [18]:
#ex-3.27
cars = ["Honda", "Toyota", "BMW", "Benz", "Nissan", "Mazda" ]
my_car = "Ford"

z = my_car in cars
print("ผลลัพธ์ของ my_car in cars =", z)
z = my_car not in cars
print("ผลลัพธ์ของ my_car not in cars =", z)

ผลลัพธ์ของ my_car in cars = False
ผลลัพธ์ของ my_car not in cars = True


## ตัวดำเนินการเอกลักษณ์ (Identity Operator)

In [ ]:
#ex-3.28
a = 5; b = 3

z = a is b
print("ผลลัพธ์ของ a is b คือ ", z)
z = a is not b
print("ผลลัพธ์ของ a is not b คือ ", z)

## ลำดับความสำคัญตัวดำเนินการ (Operator Precedence)

In [ ]:
#ex-3.29
x = 10 + 4 / 2 * 9
print ("ผลลัพธ์ของ x = 10 + 4 / 2 * 9 คือ ", x)
x = ((10 + 4) / 2) * 9
print ("ผลลัพธ์ของ ((10 + 4) / 2) * 9 คือ ", x)

# **แบบฝึกหัด**


1.   จงบอกชนิดข้อมูลจากการประกาศตัวแปรดังต่อไปนี้
```
money = 3464 
academy = "RMUTSB"
num_s = "3.241.2145
flo_num = 3241.2145
```
---
2. จงเขียนโปรแกรมคำนวณภาษีที่ต้องจ่าย โดยมีอัตราภาษีเท่ากับ 7% กำหนดให้ตัวแปรภาษีเป็นค่าคงที่ และให้ผู้ใช้งานป้อนผ่านทางคีย์บอร์ด
---
3. จงเขียนโปรแกรมคำนวณอายุ โดยให้ผู้ใช้งานป้อนข้อมูลผ่านทางคีย์บอร์ด โดยภายในคำสั่งโปรแกรมต้องมีการเรียกใช้ฟังก์ชันแปลงชนิดข้อมูลก่อนนำไปคำนวณ
---
4. จงเขียนคำสั่งโปรแกรมเพื่อหาคำตอบจากโจทย์ที่กำหนดให้
```
x = 456
print(oct(x)) =
print(hex(x)) =
print(bin(x)) =
```
---
5. จากนิพจน์ x =  125 + 315 / 3 + 5 * 2 + 15 * 4 จงหาผลลัพธ์ให้ x = 216.6666...
---
6. จากนิพจน์ x =  451 + 315 / 3 + 5 * 2 * 3 + 15 * 4 * 12 จงหาผลลัพธ์ให้ x = 1306
---


